sdk/src/main/java/io/dapr/client/DefaultDaprErrorResponseParser.java

```diff
@@ -0,0 +1,48 @@
+package io.dapr.client;
+
+import io.dapr.exceptions.DaprError;
+import io.dapr.exceptions.DaprException;
+import okhttp3.ResponseBody;
+import org.jetbrains.annotations.NotNull;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static io.dapr.client.ObjectSerializer.OBJECT_MAPPER;
+
+public class DefaultDaprErrorResponseParser implements DaprErrorResponseParser {
+
+    DaprError error;
+
+    @Override
+    public DaprException parse(@NotNull okhttp3.Response response) throws IOException {
```

Comment:
Two points here. First, can we make this interface not depend on Okhttp3? This way we can migrate to any http client in the near future.

Second point is the the http implementation will be deprecated and removed soon and gRPC will be the only implementation. Users of the SDK should not care about the protocol, except for service invocation.

So, I see two ways to proceed with this PR: make this interface protocol agnostic, making it work on with gRPC and http; Or create a new method for HTTP service invocation only and make this be used only in HTTP service invocation.

Happy to get other idea on this that would address my first two concerns.

Reply:
Thanks for your input, I've added changes to address the concerns raised, the interface works now both with HTTP and gRPC and is passed via the `DaprClientBuilder`.  Let me know if this is a good direction.